Bugfix: simplify homepage

jacqui opened the pull request:
Closes #220

This cleans up the admittedly confusing homepage code. useEffect is used very minimally now, and the props are hopefully a lot more straightforward.

Since both layouts have a "featured" article, we look that up after getting the homepage settings; then, if the settings include subfeatured top or bottom articles, we look those up. All featured articles are passed as props to the page.

I cleaned up the use of `useState` and `useEffect` and favoured using the actual properties when possible; the homepage editor makes use of those state-setting functions, so they have to remain.

I tried:

* rendering big featured story
* homepage editor, switching to large package story
* rendering large package story
* homepage editor, switching back to big featured story
* & rendering it once again

I was working against the [oaklyn api settings](https://djomoky4uky4c.cloudfront.net/graphql) but it should work against any 🤞 of the apis.

TylerFisher commented:
Generally like the look of this, though it seems to have broken the homepage editor
<img width="977" alt="Screen Shot 2021-01-27 at 2 00 01 PM" src="https://user-images.githubusercontent.com/1077075/106040107-f881e780-60a7-11eb-9928-d678a677a8c3.png">

jacqui commented:
also noticing this error on the homepage, I'll try to fix it along with this PR:

`Warning: Prop `href` did not match. Server: "politics" Client: "/politics"`

jacqui commented:
> also noticing this error on the homepage, I'll try to fix it along with this PR:
> 
> `Warning: Prop `href` did not match. Server: "politics" Client: "/politics"`

Okay I fixed the above error, and I fixed the issue with the homepage editor.

I tested with both [prod](https://d19u1w7eqbvlum.cloudfront.net/graphql) and [oaklyn](https://djomoky4uky4c.cloudfront.net/graphql) APIs.
